zeppelin-server/src/main/java/org/apache/zeppelin/rest/InterpreterRestApi.java

```diff
@@ -281,11 +300,31 @@ public Response removeRepository(@PathParam("repoId") String repoId) {
   }
 
   /**
-   * Get available types for property.
+   * Get available types for property
    */
   @GET
   @Path("property/types")
   public Response listInterpreterPropertyTypes() {
     return new JsonResponse<>(Status.OK, InterpreterPropertyType.getTypes()).build();
   }
+
+  /**
+   * Install interpreter
+   */
+  @POST
+  @Path("install")
+  @ZeppelinApi
+  public Response installInterpreter(@NotNull String message) {
+    logger.info("Install interpreter: {}", message);
+    InterpreterInstallationRequest request = gson
```

Comment:
Yes, I'm willing to do it. But personally, I don't think it's a good way because we need to keep several gson instances. BTW I'm willing to follow the current way in this time.

Reply:
What do you mean ` keep several gson instances` ? It is static instance, so should be only one instance per class

Reply:
Not a big problem. I mean I think we can reduce more of redundant gson objects, even it has only a few numbers of instances per class type. Never mind. I'll post this issue after I clarify it.